ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/services/DefaultKsqlClient.java

```diff
@@ -102,14 +103,14 @@
         .target(serverEndPoint)
         .properties(configOverrides);
 
-    final RestResponse<List<StreamedRow>> resp = getTarget(target, authHeader)
+    final RestResponse<Pair<Optional<URI>, List<StreamedRow>>> resp = getTarget(target, authHeader)
```

Comment:
I think it makes sense to have a new type of response, that consists of the debug information and the list of rows. Basically, make it a class so that we can add more debug info in the future

Reply:
It shouldn't be backwards incompatible as we introduce a new feature, no? We won't change the response for regular pull queries, especially if we introduce a per request flag.

Reply:
If we introduce a whole new endpoint (e.g. `/query_debug`), then we could change the request type and response, adding whatever new fields we want.  That seems not great since it would unnecessarily duplicate a lot of code.  And what if we introduce another mode?  We can't create the cross-product of flags, each as a new endpoint.  I would rather just have optional fields on json, where some are not set if not in debug mode, which is effectively what I did above.  The only issue is that `/query` returns an array of rows and it's not wrapped in anything.  If we wanted, we could wrap that in a response object where we could add fields, but I've held off on doing that now since I didn't require any non row-based fields.

Reply:
> The only issue is that /query returns an array of rows and it's not wrapped in anything. If we wanted, we could wrap that in a response object where we could add fields, but I've held off on doing that now since I didn't require any non row-based fields.

That's what I meant with my comment. It's fine if you want to leave it for another PR